test/toolbox_image.sh

```diff
@@ -278,12 +298,112 @@ test_execute_resize_partition_grow_rootfs_ok()
+
+    # In every line in the partition table look for a string "p3<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s|${LOOP_STORAGE_DEVICE}p3.*type|${LOOP_STORAGE_DEVICE}p3 : start=     ${USERDATA_START}, size=     ${new_userdata_size}, type|" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    execute_prepare_disk || return 0
+}
+
+test_execute_disk_prepare_grow_rootfs_ok()
+{
+    userdata_size="$((STORAGE_DEVICE_SIZE - USERDATA_START))"
+    new_userdata_size="$((userdata_size / 2))"
+    new_userdata_start="$((USERDATA_START + new_userdata_size))"
+    new_rootfs_size="$((new_userdata_start - ROOTFS_START))"
+
+    # In every line in the partition table look for a string "p2<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s|${LOOP_STORAGE_DEVICE}p2.*type|${LOOP_STORAGE_DEVICE}p2 : start=       ${ROOTFS_START}, size=     ${new_rootfs_size}, type|" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+    sed -i "s|${LOOP_STORAGE_DEVICE}p3.*type|${LOOP_STORAGE_DEVICE}p3 : start=     ${new_userdata_start}, size=     ${new_userdata_size}, type|" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    execute_prepare_disk || return 1
+    test_disk_integrity || return 1
+}
+
+test_execute_disk_prepare_grow_boot_ok()
+{
+    rootfs_size="$((USERDATA_START - ROOTFS_START))"
+    new_rootfs_size="$((rootfs_size / 2))"
+    new_rootfs_start="$((ROOTFS_START + new_rootfs_size))"
+    new_boot_size="$((new_rootfs_start + BOOT_START))"
+
+    # In every line in the partition table look for a string "p1<don't care>type" and replace it with
+    # with new the new disk partition parameters defined above.
+    sed -i "s|${LOOP_STORAGE_DEVICE}p1.*type|${LOOP_STORAGE_DEVICE}p1 : start=        ${BOOT_START}, size=     ${new_boot_size}, type|" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+    sed -i "s|${LOOP_STORAGE_DEVICE}p2.*type|${LOOP_STORAGE_DEVICE}p2 : start=     ${new_rootfs_start}, size=      ${new_rootfs_size}, type|" \
+        "${rootfs_dir}${PARTITION_TABLE_FILE}"
+
+    execute_prepare_disk || return 1
+    test_disk_integrity || return 1
+  }
+
+test_execute_disk_prepare_sha512_nok()
+{
+    partition_table_file="/tmp/partition_table"
+    sfdisk -d "${rootfs_dir}${STORAGE_DEVICE_IMG}" > "${rootfs_dir}${partition_table_file}"
+    chroot "${rootfs_dir}" sha512sum "${PARTITION_TABLE_FILE}" > "${rootfs_dir}${PARTITION_TABLE_FILE}.sha512"
+    echo "corrupted partition table data" >> "${rootfs_dir}${partition_table_file}"
+    chroot "${rootfs_dir}" "${DISK_PREPARE_COMMAND}" -t "${partition_table_file}" "${LOOP_STORAGE_DEVICE}" || return 0
+}
+
 usage()
 {
 cat <<-EOT
 	Usage:   "${0}" [OPTIONS] <file.img>
-	    -e   Stop consequtive tests on failure without cleanup
-	    -h   Print usage
+	    -e   Stop consecutive tests on failure without cleanup.
+	    -h   Print usage.
```

Comment:
lol i just removed one period to stay consistent with 'ls --help' for example, where the options are not sentences.
So what will it be, are these instructive lines (like with ls --help, docker --help etc) or ... (i can't an example)?

you choose; i'll accept both solutions :)

Reply:
I choose this